src/signer/interface.ts

```diff
@@ -14,4 +15,22 @@ export abstract class SignerInterface extends Provider {
   public abstract override addTransaction(
     transaction: Transaction
   ): Promise<AddTransactionResponse>;
+
+  /**
+   * Sign an JSON object with the starknet private key and return the signature
```

Comment:
To avoid future confusion I think the description of the method should make it clear that it is signing an off-chain message and that it will prefix the message with "StarkNet Message". Otherwise people will assume that you can also use that to sign transactions.